### DISCUSSION TITLE
Why loss increase after epoch 90th?

### DISCUSSION THREAD

**Discussion by akirayorunoe** [mrdbourke/pytorch-deep-learning#454] [Q&A]
Currently, i'm start writing the Linear Regression code. It turn out that the loss increase at 90th epoch, not same as the video,  which keep the loss of test and train the same value from epoch 70th

Code:
```
import torch
from torch import nn
import matplotlib.pyplot as plt

start =0
end=1
step=0.02
# Create data
X=torch.arange(start,end,step).unsqueeze(dim=1)
y=0.7*X+0.3

#Split train/test data (80% train, 20% test)
split_number=int(0.8*len(X))
X_train,y_train = X[:split_number],y[:split_number]
X_test,y_test = X[split_number:],y[split_number:]

def draw_plot(X_train=X_train,y_train=y_train,X_test=X_test,y_test=y_test,prediction=None):

    plt.figure(figsize=(10,7))

    plt.scatter(X_train,y_train,c='b',s=4,label='Train data')

    plt.scatter(X_test,y_test,c='g',s=4,label='Test data')

    if prediction is not None:
        plt.scatter(X_test,prediction,c='r',s=4,label='Prediction data')

    plt.legend(prop={"size": 14})
    plt.show()

class LinearRegressionModel(nn.Module):
    def __init__(self):
        super().__init__()

        self.weight=nn.Parameter(torch.rand(1,requires_grad=True))
        self.bias=nn.Parameter(torch.rand(1,requires_grad=True))

    def forward(self,x:torch.Tensor) -> torch.Tensor:
        return self.weight*x+self.bias

torch.manual_seed(42)    
#Create instance
model = LinearRegressionModel()

#Create loss and optimizer
loss_fn = nn.L1Loss()
optimizer = torch.optim.SGD(params=model.parameters(),lr=0.01)

#Train model
epochs = 100
with torch.inference_mode(): 
    y_preds_before = model(X_test)
for epoch in range(epochs):
    #Train mode
    model.train()
    #Forward prop
    y_pred=model.forward(X_train)
    #loss function
    loss = loss_fn(y_pred,y_train)
    #zero grad
    optimizer.zero_grad()
    #backprop
    loss.backward()
    #update parameters
    optimizer.step()

    #Testing
    model.eval() 
    with torch.inference_mode(): 
        #Forward pass for test
        test_preds = model(X_test)
        #loss with pred vs test y
        test_loss=loss_fn(test_preds,y_test)

    if epoch%10==0:
        print(f"Epoch {epoch} | Train loss {loss} | Test loss {test_loss}")
        print(model.state_dict())
    
with torch.inference_mode(): 
    y_preds = model(X_test)

# draw_plot(prediction=y_preds_before)
draw_plot(prediction=y_preds)

```
Result
```
...
Epoch 70 | Train loss 0.006216754671186209 | Test loss 0.014099234715104103
OrderedDict([('weight', tensor([0.6707])), ('bias', tensor([0.3120]))])
Epoch 80 | Train loss 0.002788322512060404 | Test loss 0.005826681852340698
OrderedDict([('weight', tensor([0.6878])), ('bias', tensor([0.3050]))])
Epoch 90 | Train loss 0.007095950655639172 | Test loss 0.00754010071977973
OrderedDict([('weight', tensor([0.6938])), ('bias', tensor([0.2980]))])
```
